### PR TITLE
move common dependencies to workspace root

### PR DESCRIPTION
**Pull request by ForsakenHarmony**

### CODE REVIEW

**Comment by ForsakenHarmony** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3890** <a href="https://app.graphite.dev/github/pr/vercel/turbo/3890" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/vercel/turbo/3890?utm_source=stack-comment).

File: cli/Makefile [hunk 1]
```diff
@@ -2,11 +2,14 @@ TURBO_VERSION = $(shell cat ../version.txt | sed -n '1 p')
 TURBO_TAG = $(shell cat ../version.txt | sed -n '2 p')
 
 EXT :=
-ifeq ($(OS),Windows_NT)
+FFI_LIB :=
```
Comment (ForsakenHarmony) [inline cli/Makefile:5]: need to move this to a separate PR

File: crates/turbo-malloc/Cargo.toml [hunk 2]
```diff
@@ -9,8 +9,8 @@ autobenches = false
 [lib]
 bench = false
 
-[target.'cfg(not(target_os = "linux"))'.dependencies]
-mimalloc = { version = "0.1.32", features = [], optional = true }
+[dependencies]
```
Comment (ForsakenHarmony) [inline crates/turbo-malloc/Cargo.toml:12]: I think this is more correct with the alternative below?

Comment (alexkirsz) [reply crates/turbo-malloc/Cargo.toml:12]: Not sure about this one. cc @kwonoj @Brooooooklyn

**Review by alexkirsz — APPROVED** [review]

**Comment by padmaia** [comment]
@ForsakenHarmony have you tested this with the Next.js n-api binary?

**Comment by ForsakenHarmony** [comment]
@padmaia not yet, will try

File: Cargo.toml [hunk 3]
```diff
@@ -64,20 +68,21 @@ default-members = [
   "crates/turbo-tasks-macros-shared",
   "crates/turbo-tasks-memory",
   "crates/turbo-tasks-testing",
-  "crates/turbo-tasks",
+  "crates/turbopack",
```
Comment (NicholasLYang) [inline Cargo.toml:71]: this is a duplicate

**Review by NicholasLYang — APPROVED** [review]
Looks good to me. No real objections as long as CI passes

File: Cargo.toml (hunk 3 above)
Comment (ForsakenHarmony) [reply Cargo.toml:71]: good catch
